bin/build_cmd.ml

```diff
@@ -1,15 +1,12 @@
 open Stdune
 open Import
 
-let run_build_system ?report_error ~common
-    ~(targets : unit -> Target.t list Memo.Build.t) () =
+let run_build_system ?report_error ~common ~(request : unit Action_builder.t) ()
```

Comment:
Just a thought: maybe `Action_builder.t` could be called `Request.t`, since this is how we call it here, at the top level.

Reply:
(The current name `Action_builder` irks me every time I see it, so I volunteer to do the renaming if we come up with a better name!)

Reply:
"requesting" things to be built is by far a minority use case for Action_builder.t and doesn't even use most of its power. The purpose of Action_builder is to construct `Action.t`s along with their deps.

I'm all for improving the name, but I'm against changing it in the other direction. :-)

Reply:
I agree that `Action_builder`'s API is pretty specialised to constructing actions.

However, the type `Action.t Request.t` seems to express the notion of constructing an `Action.t` clearly enough. And, more generally, a value of type `'a Request` would mean a "request" to construct an `'a`, which might involve running some actions and incurring dependencies.

Grepping Dune sources, instantiating `'a` to `Action.t` is not the most popular type. `'a = unit` occurs a lot more often!

Reply:
Here are some figures from Dune main:

* 4 occurrences of `Action.t Action_builder.t`
* 11 occurrences of `'a Action_builder.t`
* 20 occurrences of `unit Action_builder.t`
* 45 occurrences of `'a list Action_builder.t`, with the most popular choice of `'a` being `string`.

So, really, , if we look at the type annotations, constructing actions is in the tiny minority :) Most of the time we read/write types like `string list Action_builder.t` which do not seem very helpful.

Reply:
"requesting" to construct a `unit` makes even less sense to me, to be honest.
I think mentioning "deps" would be nice, since declaring deps is *the* internally-observable action of this monad (the rest is simply computation). 
For example `With_deps` immediately comes to mind, or `Build_with_deps` to express the "lazy" nature.
In jenga we've been calling this a `Dep` monad, and I like that short name. Too bad the name collides with the static description of an individual dep.

Reply:
The strings and units are usually partial command lines and descriptions of certain aspects of behavior of a certain action, though, so I don't understand the point you're making. Of course action will be in the minority because to construct one action you need a lot of strings and units.

Reply:
Agreed, I like `Dep` too. Maybe `Arg.Dep.t` should be renamed to `Arg.Goal.t`, unlocking the name for the `Dep` monad.

Reply:
I was more thinking about the elephant that is `Dune_engine.Dep.t`.

Reply:
Yeah, I like `Dep` as well. Maybe we can find another name for the current `Dep`

Reply:
Oops, I didn't notice it :) This elephant is pretty well concealed, there are just a few use sites, so it should be easy to rename too.

> The strings and units are usually partial command lines and descriptions of certain aspects of behavior of a certain action, though, so I don't understand the point you're making. 

My point is that local snippets of code like `Action_builder.return 8` make little sense to me. How does it build an action? And 99% of code involving this monad is like this. Most (although not all) of these values will indeed eventually end up as part of an action but that just doesn't seem important or interesting to me. The fact that you can attach dependencies to these values is interesting and important.